src/Donations/Models/Donation.php

```diff
@@ -79,7 +80,7 @@ class Donation extends Model implements ModelCrud, ModelHasFactory
         'subscriptionId' => ['int', 0],
         'billingAddress' => BillingAddress::class,
         'anonymous' => ['bool', false],
-        'levelId' => ['int', 0],
+        'levelId' => ['string', ''],
```

Comment:
@JasonTheAdams the `levelId` or what used to be called a form's `price-id` is basically just the key we refer to a donation level.  The legacy type is `string|int` because it can be saved as "custom" or an integer - which just reflects the array position they were in when the form was created.  This method of array positions dictating the identifier has caused some issues in the past. 

After discussing with @ravinderk we concluded that in the future, it would make more sense if these id's were stored and retrieved with a more meaningful key string that would be generated by the donation level label or amount. So as far as the model is concerned, the `levelId` should be a string that should reflect the donation level.

Reply:
Roger that! Thanks for the thorough response. 👍